src/transform/src/lib.rs

```diff
@@ -681,33 +713,27 @@ impl Optimizer {
                 name: "fixpoint_physical_01",
                 limit: 100,
                 transforms: vec![
-                    Box::new(column_knowledge::ColumnKnowledge::default()),
-                    Box::new(fold_constants::FoldConstants {
-                        limit: Some(FOLD_CONSTANTS_LIMIT),
-                    }),
-                    Box::new(canonicalization::ReduceScalars),
-                    Box::new(demand::Demand::default()),
-                    Box::new(literal_lifting::LiteralLifting::default()),
+                    Box::new(ColumnKnowledge::default()),
+                    Box::new(fold_constants_fixpoint()),
+                    Box::new(Demand::default()),
+                    Box::new(LiteralLifting::default()),
                 ],
             }),
-            Box::new(literal_constraints::LiteralConstraints),
+            Box::new(LiteralConstraints),
             Box::new(Fixpoint {
                 name: "fixpoint_join_impl",
                 limit: 100,
-                transforms: vec![Box::new(join_implementation::JoinImplementation::default())],
+                transforms: vec![Box::new(JoinImplementation::default())],
             }),
-            Box::new(canonicalize_mfp::CanonicalizeMfp),
+            Box::new(CanonicalizeMfp),
             // Identifies common relation subexpressions.
             Box::new(cse::relation_cse::RelationCSE::new(false)),
-            Box::new(fold_constants::FoldConstants {
-                limit: Some(FOLD_CONSTANTS_LIMIT),
-            }),
-            Box::new(canonicalization::ReduceScalars),
+            Box::new(fold_constants_fixpoint()),
             // We need this to ensure that `CollectIndexRequests` gets a normalized plan.
             // (For example, `FoldConstants` can break the normalized form by removing all
             // references to a Let, see https://github.com/MaterializeInc/database-issues/issues/6371)
-            Box::new(normalize_lets::NormalizeLets::new(false)),
-            Box::new(typecheck::Typecheck::new(ctx.typecheck()).disallow_new_globals()),
+            Box::new(NormalizeLets::new(false)),
```

Comment:
Very minor, but I think this `NormalizeLets` is now rendundant, because `fold_constants_fixpoint()` just above ended with it. That might be great news, in that it means we are a bit closer to having some expectations about the resulting state (the closer we get to ending with something like "canonicalization" the better, imo).